homeassistant/components/pterodactyl/api.py

```diff
@@ -88,20 +84,26 @@ async def async_get_data(self) -> dict[str, PterodactylData]:
         """Update the data from all Pterodactyl servers."""
         data = {}
 
-        if self.pterodactyl is None:
-            raise PterodactylNotInitializedError(
-                "Pterodactyl API is not initialized yet"
-            )
-
-        try:
-            for identifier in self.identifiers:
+        for identifier in self.identifiers:
+            try:
                 server = await self.hass.async_add_executor_job(
-                    self.pterodactyl.client.servers.get_server, identifier
+                    self.pterodactyl.client.servers.get_server,  # type: ignore[union-attr]
+                    identifier,
                 )
                 utilization = await self.hass.async_add_executor_job(
-                    self.pterodactyl.client.servers.get_server_utilization, identifier
+                    self.pterodactyl.client.servers.get_server_utilization,  # type: ignore[union-attr]
+                    identifier,
                 )
-
+            except (
+                PydactylError,
+                BadRequestError,
+                PterodactylApiError,
+            ) as error:
+                raise PterodactylConnectionError(error) from error
+            except Exception as error:
```

Comment:
Ideally don't catch bare exceptions out of the config flow

Reply:
I removed catching bare exceptions in api.py, but added it to config_flow.py instead. I fear that I have to catch way more exceptions I don't know of after first users try out the integration 😄